### PR TITLE
Keep version and language on redirect

### LINKED ISSUE #1025: Page redirects lead to short url, not the canonical one
Hello, I changed my documentation a bit, and these changes require me to make some redirects to respect the old links. So I've made the following page redirect in my admin panel:

```
users-guide/getting-started/installation.html
getting-started/installation.html
```

Redirect itself is working fine, but leads to another domain, non-canonical. For example, redirect descibed above leads from http://docs.hangfire.io/en/latest/users-guide/getting-started/installation.html to http://hangfire.readthedocs.org/en/latest/getting-started/installation.html. But I want docs to be redirected to the *docs.hangfire.io* domain instead of *hangfire.readthedocs.org*.

How to make redirect respect my canonical domain?

### PR DESCRIPTION
**Pull request by gregmuellegger**
The first comment on #1025 mentions that the version is not kept for redirects.

This PR keeps the version and the language on redirects. It also features a nice refactor for better code structure and moving more logic into the Redirect model. Keeping the logic where the configuration is.

Fixes #1025

### CODE REVIEW

**Comment by gregmuellegger** [comment]
This is now ready for review.

File: readthedocs/redirects/models.py [hunk 1]
```diff
@@ -67,3 +78,97 @@ def __unicode__(self):
...
+            if version_slug is None:
+                version_slug = self.project.get_default_version()
+            else:
+                versions = self.project.versions.all()
+                if not versions.filter(slug=version_slug).exists():
+                    version_slug = self.project.get_default_version()
+            url_kwargs.update({
+                'lang_slug': language,
+                'version_slug': version_slug,
+            })
```
Comment (ericholscher) [inline readthedocs/redirects/models.py]: I feel like we keep rewriting this bit of code throughout the codebase, and should standardize on one approach and centralize it. It's in at least 3 places (core/views.py, project model get_docs_url, and redirects). I have started cleaning up the doc serving & subdomain_urls to standardize them in the subproject branch, and I think we should be able to pull this into one place.

Comment (gregmuellegger) [inline readthedocs/redirects/models.py]: So true. That should be centralized somehow. When do you think will the subproject-alias branch land in master? I would argue for going down the road of having separate implementations for now and merging them together when this branch and subproject-alias both have hit master.

Comment (ericholscher) [inline readthedocs/redirects/models.py]: Agreed.

**Pull request merged**